homeassistant/util/__init__.py

```diff
@@ -53,7 +53,7 @@ def repr_helper(inp: Any) -> str:
     return str(inp)
 
 
-def convert(value: T, to_type: Callable[[T], U],
+def convert(value: Optional[T], to_type: Callable[[T], U],
```

Comment:
Why is that Optional?

Reply:
Because it is really optional -- the code has specific checks and treatment of `None` values.